src/wallet/wallet.cpp

```diff
@@ -2042,6 +2048,9 @@ bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, int nConfMine, int
         if (output.nDepth < (pcoin->IsFromMe(ISMINE_ALL) ? nConfMine : nConfTheirs))
             continue;
 
+        if (pcoin->InMempool() && pcoin->GetMempoolAncestorCount() > nMaxAncestors)
```

Comment:
Perhaps worth a comment here that we're assuming `AvailableCoins()` has already filtered out any unconfirmed and not-in-mempool coins?